### DISCUSSION TITLE
Continue forward simulation

### DISCUSSION THREAD

**Discussion by PengWei97** [idaholab/moose#16447] [Q&A General] [answered]
Dear MOOSE experts,

I would like to ask, can moose continue to simulate forward at the time of the previous simulation instead of re-simulating?
In detail, if I set the time as follows:
`  start_time = 0.0
  end_time = 8000
`

After the simulation is completed, I set ent_time = 10000, but hope that start_time = 8001. Does moose have such a function?
If so, how can I do it?

Any suggestions or recommendations to fix these problems would be greatly appreciated.
Thank you
Wei Peng

**Comment by dschwen** [comment, marked as answer]
You need to add checkpoint = true in your Outputs block in your first run and then start your second run with --recover after increasing the end time.

**Reply by PengWei97** [reply]
Thank you for your reply, the problem is solved. 
Follow your suggestion, Following your suggestion, I solved the problem in two steps:
First add "checkpoint = true" to the outputs block in the input file. 
After the simulation is complete, modify the end_time, and finally run the name command line "*-opt -i *.i --recover".

**Reply by PengWei97** [reply]
Dear dschwen,
Regarding the function of this moose, I still have some questions and hope to get your suggestion.
First, I use recover to run grain_growth_2D_graintracker.i, it can continue to simulate forward. @dschwen 

But when I change the input file to poly_grain_growth_2D_eldrforce.i, it shows an error in the terminal, the error is as follows:
```
*** ERROR ***
ERROR: Neither one of the following files can be located:
'_mesh.cpr/1/header.cpr' nor
'_mesh.cpr'
If you are running a parallel job, double check that you've created a split for 1 ranks.
Note: One of paths above may refer to a valid directory on your system, however we are attempting to read a valid header file.
```
For the coupled model of multiple physics modules, do I need to perform other operations to continue the forward simulation?

**Comment by WilkAndy** [comment]
Have a read of https://mooseframework.inl.gov/application_usage/restart_recover.html to see if it satisfies your requirements.

**Reply by PengWei97** [reply]
Thank you for your quick reply, the webpage you gave is very important to me.
